### PR TITLE
Feature/toast notification

### LINKED ISSUE #80: Toast Library and implement

### PR DESCRIPTION
**Pull request by codyenokida**
Fixes #80 

Toast notifications for transactions and approvals!

Notable changes:
1. installed react-hot-toast dependency
2. Two types of different toasts: TransactionToasts and Generic Toast
3. TransactionToast for all transactions and approval requests
4. Generic Toast for more generic error or warning messages.
2. Created a ToastController that exports functions to render Toasts
3. Created storybooks

Feedback appreciated!

### CODE REVIEW

**Comment by gpxl-dev** [comment]
I haven't looked in huge detail at implementation yet, but I think this mostly needs to be discussed with the designers / someone with design authority, but there are a few things that differ a bit from designs (as well as general approach to toasts as per my thoughts on [discord](https://discord.com/channels/590643190281928738/832210263284383764/869163459361779782) :

1. Generic toast is much bigger than error message popup in design (font size, wise).
2. Close icon vertical centering on generic toast
3. Generic toast appears to be able to take some supplementary information (e.g. an error code), which isn't accounted for in its props currently.
4. Transaction toasts are fixed width in the designs

One implementation point I'm not sure about is that the start time is passed formatted to the toasts. I think this is a slightly jarring developer experience as, when I implement a toast, I need to reference the formatting on all other toasts, and similarly if I want to change the way the start time is displayed on a toast, I need to do it in many places. I would pass a javascript millisecond timestamp (i.e. `Date.now()`) and do the formatting in the toast itself.

Reference images:

Storybook tx toast:
image

Design tx toast:
image

Storybook generic toast: 
image

Design generic toast (not to scale, sorry):
image

Relative sizing of toasts on design:

image

File: src/components/Toasts/GenericToast.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+const GenericToast = ({
+  onClose,
+  type,
+  intent = "error",
+}: GenericToastProps) => {
+  const { t } = useTranslation(["toast", "common"]);
+
+  return (
+    <Container intent={intent}>
+      <BiInfoCircle style={{ width: "1.5rem", height: "1.5rem" }} />
```
Comment (piersss) [inline src/components/Toasts/GenericToast.tsx:31]: Move styling to GenericToast.styles.tsx

File: src/components/Toasts/GenericToast.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <TextContainer>
+        <strong>
+          {intent === "error" ? t("common:error") : t("common:warning")}
+        </strong>
+        : {t(type)}
+      </TextContainer>
+      <HiXContainer>
+        <HiX
+          style={{
```
Comment (piersss) [inline src/components/Toasts/GenericToast.tsx:40]: This should be a button, not a div, and move styling to GenericToast.styles.tsx

File: src/components/Toasts/GenericToast.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  grid-template-columns: auto auto auto;
+  padding: 1rem;
+`;
+
+export const InfoContainer = styled.div<{ theme: ThemeProps }>`
+  display: grid;
+  grid-template-columns: auto auto auto;
+`;
+
+export const HiXContainer = styled.div<{ theme: ThemeProps }>``;
```
Comment (piersss) [inline src/components/Toasts/GenericToast.styles.tsx:23]: Vertically center the close button like so:

```
display: flex;
align-items: center;
```

File: src/components/Toasts/ToastController.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import toast from "react-hot-toast";
+import TransactionToast from "./TransactionToast";
+import GenericToast from "./GenericToast";
+
+const formatDate = (date: Date) => {
```
Comment (piersss) [inline src/components/Toasts/ToastController.tsx:5]: `const formatDate = (date: Date): string => {`

Move this function to Toasts/helpers/formatDate.ts and write some tests in Toasts/helpers/formatDate.test.ts

File: src/components/Toasts/Toaster.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Toaster as T } from "react-hot-toast";
+import styled from "styled-components/macro";
+
+const ToasterWrapper = styled.div`
```
Comment (piersss) [inline src/components/Toasts/Toaster.tsx:4]: Move styles to Toaster.styles.tsx

File: src/components/Toasts/TransactionToast.styles.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import styled from "styled-components/macro";
+
+export const Container = styled.div`
+  color: ${(props) => props.theme.colors.white};
```
Comment (piersss) [inline src/components/Toasts/TransactionToast.styles.tsx:4]: Try to order your styling according to the box model. For instance `color` and `background` don't have any effect on the size / position of your, but `display` and `grid-template-columns` do.

```
  display: grid;
  grid-template-columns: auto auto auto;
  grid-gap: 1rem;
  justify-items: center;
  align-items: center;
  padding: 1rem;
  color: ${(props) => props.theme.colors.white};
  background: ${(props) => props.theme.colors.black};
```

File: src/components/Toasts/TransactionToast.styles.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  justify-items: center;
+  align-items: center;
+`;
+
+type IconContainerProps = {
+  error?: boolean;
+};
+
+export const IconContainer = styled.div<IconContainerProps>`
+  background: ${(props) =>
```
Comment (piersss) [inline src/components/Toasts/TransactionToast.styles.tsx:19]:
```
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  width: 2rem;
  height: 2rem;
  color: white;
  background: ${(props) =>
    props.error ? props.theme.colors.red : props.theme.colors.primary};

```

**Review by piersss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**